util/drakeGeometryUtil.cpp

```diff
@@ -906,6 +906,28 @@ void rpydot2angularvelMatrix(const Eigen::MatrixBase<DerivedRPY>& rpy,
   }
 }
 
+template<typename DerivedRPY, typename DerivedRPYdot, typename DerivedOMEGA>
+void rpydot2angularvel(const Eigen::MatrixBase<DerivedRPY>& rpy,
+                       const Eigen::MatrixBase<DerivedRPYdot>& rpydot,
+                             Eigen::MatrixBase<DerivedOMEGA>& omega,
+                             typename Gradient<DerivedOMEGA,RPY_SIZE,1>::type* domega)
+{
+  EIGEN_STATIC_ASSERT_VECTOR_SPECIFIC_SIZE(Eigen::MatrixBase<DerivedRPY>, RPY_SIZE);
+  EIGEN_STATIC_ASSERT_VECTOR_SPECIFIC_SIZE(Eigen::MatrixBase<DerivedRPYdot>, RPY_SIZE);
+  EIGEN_STATIC_ASSERT_MATRIX_SPECIFIC_SIZE(Eigen::MatrixBase<DerivedOMEGA>, RPY_SIZE, 1);
+
+  Matrix3d E;
```

Comment:
One more thing that I didn't tell you about: we'd like to support different scalar types (not just double, but also Polynomial, AutoDiffScalar, TrigPoly, ...). Here you're assuming that the output type should have double as the scalar type. The right way to do it is something like:
```
typedef DerivedRPY::Scalar Scalar; // the actual scalar type used in the input matrix
Matrix<Scalar, 3, 3> E;
// ... snip ...
Matrix<Scalar, 9, 3> dE; // alternatively, Gradient<Matrix<Scalar, 3, 3>, 3>::type dE or even Gradient<decltype(E), 3>::type dE
// ... snip ...
```